src/BizHawk.Client.Common/movie/tasproj/ZwinderStateManager.cs

```diff
@@ -276,10 +276,9 @@ private void AddToReserved(ZwinderBuffer.StateInformation state)
 				return;
 			}
 
-			var bb = new byte[state.Size];
-			var ms = new MemoryStream(bb);
+			var ms = new MemoryStream();
 			state.GetReadStream().CopyTo(ms);
-			_reserved.Add(state.Frame, bb);
+			_reserved.Add(state.Frame, ms.ToArray());

```

Comment:
This is regrettable, although I understand why you did it.

Reply:
The previous logic crashes on 2.8 too fwiw (I guess showing people don't use this compression in tastudio).

Reply:
The size used for creating the array is the compressed size, and reading the stream like CopyTo does will read out the decompressed size, and memory streams init with an array cannot be resized.

Reply:
Right, I get that.  Does the current live code actually allow you to select compression for the state manager?

Reply:
The current code allows for selecting compression or no compression. In rewind menu it's "zlib compression" (reworded in this PR as "zstd compression"). In TAStudio it's in the "State History Settings." Delta compression is a completely separate rewinder.

Reply:
So it's busted, hmm hmm.

The delta rewinder doesn't allow the kind of access pattern that tarsestudio needs, so yeah, that's a nonstarter.